src/Workspaces/Core/Portable/Execution/Serializer.cs

```diff
@@ -88,6 +190,49 @@ public T Deserialize<T>(string kind, ObjectReader reader, CancellationToken canc
             }
         }
 
+        private string GetLanguageName(object value)
+        {
+            // for given object, we need to figure out which language the object belong to. 
+            // we can't blindly get language service since that will bring in language specific dlls.
+            foreach (var languageName in _workspaceServices.SupportedLanguages)
+            {
+                IOptionsSerializationService service;
+                if (_lazyLanguageSerializationService.TryGetValue(languageName, out service))
+                {
+                    if (service.Owns(value))
+                    {
+                        return languageName;
+                    }
+
+                    continue;
+                }
+
+                // this should be only reached once per language value actually belong to
+                var mefWorkspaceServices = _workspaceServices as MefWorkspaceServices;
+                if (mefWorkspaceServices != null)
+                {
+                    MefLanguageServices languageServices;
+                    if (!mefWorkspaceServices.TryGetLanguageServices(languageName, out languageServices))
+                    {
+                        // this is a bit fragile since it depends on implementation detail but there is no other way
+                        // to figure out which language a type belong to without loading other languages
```

Comment:
sure like I said, if compiler team doesn't pick it up, I will start work. the reason I want compiler team to pick it up is, I don't serialize everything compilation option has. there are a few inputs I ignore such as resolvers. once it moves down to compiler layer and become public API, I believe it needs to support all those inputs.

all complication comes from the fact those options are not data but contains behavior in them that is depends on host which VS version provide one that depends on VS.